### ISSUE TITLE
cant start Veil

### ISSUE THREAD

**Issue by slavadba** [Veil-Framework/Veil#195]
## Veil version
dont now because cant start veil - see error below, but I think it should be latest - I tried install Veil yesterday.

## OS Used - all info (architecture, linux flavor, etc)
 Linux kali 4.14.0-kali3-amd64 #1 SMP Debian 4.14.12-2kali1 (2018-01-08) x86_64 GNU/Linux

## How did you install Veil? (Apt, Clone from Github, etc.)
apt -y install veil
veil --setup

## Did you run the setup script?
yes. 

## Pastebin link to error you are encountering (include console actions you took prior to error)
root@kali:/usr/share/veil# ./Veil.py 
Traceback (most recent call last):
  File "./Veil.py", line 95, in <module>
    the_conductor = orchestra.Conductor(args)
  File "/usr/share/veil/lib/common/orchestra.py", line 23, in __init__
    self.load_tools(cli_stuff)
  File "/usr/share/veil/lib/common/orchestra.py", line 65, in load_tools
    name.replace("/", ".").rstrip('.py'), name)
  File "/usr/lib/python3.6/imp.py", line 172, in load_source
    module = _load(spec)
  File "<frozen importlib._bootstrap>", line 684, in _load
  File "<frozen importlib._bootstrap>", line 665, in _load_unlocked
  File "<frozen importlib._bootstrap_external>", line 678, in exec_module
  File "<frozen importlib._bootstrap>", line 219, in _call_with_frames_removed
  File "Tools/Evasion/Tool.py", line 16, in <module>
    from Tools.Evasion.evasion_common import shellcode_help
  File "/usr/share/veil/Tools/Evasion/evasion_common/shellcode_help.py", line 30, in <module>
    import Tools.Ordnance.Tool as Ordnance_Import
AttributeError: module 'Tools.Ordnance' has no attribute 'Tool'
root@kali:/usr/share/veil#

## Expected behavior
start veil

## Any additional info you want to tell me

**Comment by g0tmi1k** [comment]
Once #189 is merged, I suggest re-trying.

**Comment by slavadba** [comment]
"I suggest re-trying" - its means - remove Veil dir and make the installation from the very beginning?
Or I can instrall only changes by running several commands ( and if yes could you suggest these commands please )?

**Comment by g0tmi1k** [comment]
You said:
```
How did you install Veil? (Apt, Clone from Github, etc.)
apt -y install veil
veil --setup
```

So re-trying for you would be:
```
apt --purge remove -y veil
apt install -y veil
veil
```

**Comment by n1ghty93** [comment]
Same Problem here.
Git and APT everything is throwing the same errors. Any workarounds?
Kali everything updated and installed

**Comment by g0tmi1k** [comment]
@n1ghty93 Until #189 is merged in, either:

- Wait.
- Manually pull in the changes in from #189.

**Comment by n1ghty93** [comment]
So ill need to purge veil, clone the git? or isnt it in there, too?

**Comment by g0tmi1k** [comment]
Best just to wait.

**Comment by n1ghty93** [comment]
I would but i cant i need it for work :D

**Comment by slavadba** [comment]
I reinstall Veil using this commands
apt --purge remove -y veil
apt install -y veil
veil

but the error  the same:
.
.
.
[*] Settings already detected... Skipping...


 [I] If you have any errors running Veil-Evasion, delete the Veil Wine profile ('rm -rf /root/.config/wine/veil') and re-run: '/usr/share/veil/setup/setup.sh -c'


 [I] Done!

Traceback (most recent call last):
  File "./Veil.py", line 95, in <module>
    the_conductor = orchestra.Conductor(args)
  File "/usr/share/veil/lib/common/orchestra.py", line 23, in __init__
    self.load_tools(cli_stuff)
  File "/usr/share/veil/lib/common/orchestra.py", line 65, in load_tools
    name.replace("/", ".").rstrip('.py'), name)
  File "/usr/lib/python3.6/imp.py", line 172, in load_source
    module = _load(spec)
  File "<frozen importlib._bootstrap>", line 684, in _load
  File "<frozen importlib._bootstrap>", line 665, in _load_unlocked
  File "<frozen importlib._bootstrap_external>", line 678, in exec_module
  File "<frozen importlib._bootstrap>", line 219, in _call_with_frames_removed
  File "Tools/Evasion/Tool.py", line 16, in <module>
    from Tools.Evasion.evasion_common import shellcode_help
  File "/usr/share/veil/Tools/Evasion/evasion_common/shellcode_help.py", line 30, in <module>
    import Tools.Ordnance.Tool as Ordnance_Import
AttributeError: module 'Tools.Ordnance' has no attribute 'Tool'
root@kali:~#

**Comment by g0tmi1k** [comment]
Merged!
Could you update Veil and re-test with v3.1.6?

**Comment by slavadba** [comment]
I made 

apt --purge remove -y veil
apt install -y veil
veil

and got the same mistakes. 

questions: on which OS version 3.1.6 solved the problems described in this topic?
why you can not download Kali, reproduce the problem - and repair it?

**Comment by g0tmi1k** [comment]
The package in Kali is `v3.1.1`
So until the package is updated, installing via `apt-get` will always give the same error. The Kali package should hopefully be updated in the next week or so.
The Kali package is out of @ChrisTruncer 's hand. Which is why he can't just repair it.
And why there isn't an OS you can download with it all setup.

You need to use `git clone` to install the latest (at this time of writing, `v3.1.6`). See the README for how.

Options are:
- Wait for Kali's package to update ~ https://pkg.kali.org/pkg/veil
- Install via `git clone`

**Comment by slavadba** [comment]
I install using git clone  - and its fix the issue, thank you very much!

**Comment by dela-dee** [comment]
i am getting this error whenever i run veil can someone assist me. 


Traceback (most recent call last):
  File "./Veil.py", line 16, in <module>
    from lib.common import helpers
  File "/usr/share/veil/lib/common/helpers.py", line 22, in <module>
    if not os.path.exists( settings.GOLANG_PATH ):
AttributeError: module 'settings' has no attribute 'GOLANG_PATH'

**Comment by g0tmi1k** [comment]
And thats off-topic @dela-dee . Need todo a new issue

**Comment by jaaabir** [comment]
> i am getting this error whenever i run veil can someone assist me.
> 
> Traceback (most recent call last):
> File "./Veil.py", line 16, in
> from lib.common import helpers
> File "/usr/share/veil/lib/common/helpers.py", line 22, in
> if not os.path.exists( settings.GOLANG_PATH ):
> AttributeError: module 'settings' has no attribute 'GOLANG_PATH'

--------------------------------------------------------------------------------------------------------------

for those of you who have that above error for veil then the solution is to manually install go by following the steps from https://tzusec.com/how-to-install-golang-in-kali-linux/ . 

Then after successfully installing it,locate where the veil.py , helpers.py and settings.py 
(i guess they all will be in veil dir itself) .

next go to settings.py dir and nano settings.py and then enter 

GOLANG_PATH = "path to the go lang"

finally save all those 3 py files.

maybe once again when you run veil it may show error and if it did enter the command 

/usr/share/veil/config/setup.sh --force --silent

and after this setup it'll work once you run veil again.

**Comment by AdityaRed** [comment]
Traceback (most recent call last):
  File "/opt/Veil/./Veil.py", line 10, in <module>
    from lib.common import orchestra
  File "/opt/Veil/lib/common/orchestra.py", line 7, in <module>
    import readline
ModuleNotFoundError: No module named 'readline'

I am getting this error whenever I try to run veil
